pkg/factory/risingwave_object_factory.go

```diff
@@ -857,6 +857,11 @@ func (f *RisingWaveObjectFactory) buildPodTemplate(component, group string, podT
 
 		// Set the tolerations.
 		podTemplate.Spec.Tolerations = append(podTemplate.Spec.Tolerations, groupTemplate.Tolerations...)
+
+		// Set the priority value
+		if groupTemplate.Priority != nil {
+			podTemplate.Spec.Priority = groupTemplate.Priority
```

Comment:
I think we should run a deep copy to avoid unexpected sharing of the pointer. Try to use `pointer.Int32(*groupTemplate.Priority)`.